src/components/transactions/TxFilterForm/index.tsx

```diff
@@ -0,0 +1,236 @@
+import Paper from '@mui/material/Paper'
+import Grid from '@mui/material/Grid'
+import FormControl from '@mui/material/FormControl'
+import RadioGroup from '@mui/material/RadioGroup'
+import FormLabel from '@mui/material/FormLabel'
+import FormControlLabel from '@mui/material/FormControlLabel'
+import Radio from '@mui/material/Radio'
+import TextField from '@mui/material/TextField'
+import Button from '@mui/material/Button'
+import { Controller, FormProvider, useForm, useFormState, type DefaultValues } from 'react-hook-form'
+import { useMemo, type ReactElement } from 'react'
+
+import AddressBookInput from '@/components/common/AddressBookInput'
+import DatePickerInput from '@/components/common/DatePickerInput'
+import { validateAmount } from '@/utils/validation'
+import { trackEvent } from '@/services/analytics'
+import { TX_LIST_EVENTS } from '@/services/analytics/events/txList'
+import { TxFilterType, txFilter, useTxFilter, TxFilter } from '@/utils/tx-history-filter'
+
+export enum TxFilterFormFieldNames {
+  FILTER_TYPE = 'type',
+  DATE_FROM = 'execution_date__gte',
+  DATE_TO = 'execution_date__lte',
+  RECIPIENT = 'to',
+  AMOUNT = 'value',
+  TOKEN_ADDRESS = 'token_address',
+  MODULE = 'module',
+  NONCE = 'nonce',
+}
+
+export type TxFilterFormState = {
+  [TxFilterFormFieldNames.FILTER_TYPE]: TxFilterType
+  [TxFilterFormFieldNames.DATE_FROM]: string | null | Date
+  [TxFilterFormFieldNames.DATE_TO]: string | null | Date
```

Comment:
How can it be both a string and a date?

Reply:
`<DatePicker>` can accept a `defaultValue` of a `string`. It outputs a `Date` or `null` if the user enters/doesn't enter a value.